### DISCUSSION TITLE
make specials treatments that display image

### DISCUSSION THREAD

**Discussion by nico1080** [mattblovell/kodi_panel#20] [Ideas]
The `codec `special treatment will allow to display custom name for each codec, but it is still plain text.
It would be nice to display the logo of the codec (I always liked their graphism) 
Few example of logos from wikipedia:
![image](https://user-images.githubusercontent.com/5569758/104955315-5472af00-59ca-11eb-9ef8-9d933c97112a.png)![image](https://user-images.githubusercontent.com/5569758/104955424-8be15b80-59ca-11eb-9737-db43cdc504dc.png)

Same kind of treatment could apply to audio/video codec, video resolution, audio bitpersample/samplerate, etc.

**Comment by mattblovell** [comment]
This is certainly possible, but I do see two immediate challenges:
- For proper display, the images may have to be resized for a particular display resolution.  So long as the image source is sufficiently large, scaling down isn't a problem (although it will consume compute cycles).  Scaling *up* in size could be problematic.  
- I've never looked into the distribution rights for such logos.  If Wikipedia is willing to host them, though, I guess that isn't much of an issue.

If the feature is entirely under end-user control, and the end-user has to provide the images to use, that would largely address the concerns above.

**Reply by nico1080** [reply]
> * .  So long as the image source is sufficiently large, scaling down isn't a problem (although it will consume compute cycles).  Scaling _up_ in size could be problematic.

Regarding compute cycle there could be a script that:

1. Resize the image and store it in a cache folder  the first time the image is called
2. On later calls the script would call the resized image

The resizing script would also need to have option regarding aspect ratio (keep original, fill the whole picture, etc) and maybe quality.

I imagined that feature to be entirely under end-user control

**Comment by mattblovell** [comment]
Upon further thought, making use of a dictionary lookup table for Idea #16 would also permit the above to be implemented.

**Comment by mattblovell** [comment]
Should be possible with v1.22

One would delete the existing 'codec' entry in the `STRING_CB` table/dictionary and add some customized function to `ELEMENT_CB`.  

Alternatively, you can just leave the string manipulating 'codec' entry in place and come up with some completely *new* name for the image-drawing codec lookup.  You would then use that new name within the fields array of whatever layout you wanted.

**Comment by nico1080** [comment]
I made it for audio and video codec. Right now I am using confluence logo, but I will look for nicer logos.
I will also make the same for video resolution (1080p etc)

My commits in my fork: 5cdf5b664724d3b8f51fea32888e5972857a7e5a 731143e4cd585b7bb1bb4827aa47ff179ea15d0e and 5dd398c3d253be1c27e356c753a2a3f6945797b9
